galaxy_ng/app/migrations/0026_move_perms_to_roles.py

```diff
@@ -0,0 +1,136 @@
+    ), "galaxy.collection_admin"),
+    ((
+        ("container", "delete_containerrepository"),
+        ("container", "namespace_change_containerdistribution"),
+        ("container", "namespace_modify_content_containerpushrepository"),
+        ("container", "namespace_push_containerdistribution"),
+        ("container", "add_containernamespace"),
+        ("container", "change_containernamespace"),
+    ), "galaxy.execution_environment_admin"),
+    ((
+        ("ansible", "modify_ansible_repo_content"),
+    ), "galaxy.content_admin"),
+    ((
+        ("galaxy", "add_namespace"),
+        ("galaxy", "change_namespace"),
+        ("galaxy", "delete_namespace"),
+        ("galaxy", "view_namespace"),
+        ("galaxy", "upload_to_namespace"),
+        ("ansible", "delete_collection"),
+    ), "galaxy.namespace_owner"),
+    ((
+        ("galaxy", "upload_to_namespace"),
+        ("ansible", "delete_collection"),
+    ), "galaxy.publisher"),
+    ((
+        ("galaxy", "view_group"),
+        ("galaxy", "delete_group"),
+        ("galaxy", "add_group"),
+        ("galaxy", "change_group"),
+    ), "galaxy.group_admin"),
+    ((
+        ("galaxy", "view_user"),
+        ("galaxy", "delete_user"),
+        ("galaxy", "add_user"),
+        ("galaxy", "change_user"),
+    ), "galaxy.user_admin"),
+    ((
+        ("galaxy", "add_synclist"),
+        ("galaxy", "change_synclist"),
+        ("galaxy", "delete_synclist"),
+        ("galaxy", "view_synclist"),
+        ("ansible", "change_collectionremote"),
+    ), "galaxy.synclist_owner"),
+]
+
+
+def locked_perm_in_permissions(locked_perm, permissions):
+    # locked_perm: Permission object
+    # permissions: queryset of Permission objects
+    for perm in permissions:
+        if perm.id == locked_perm.id:
+            return True
+    return False
+
+
+def remove_locked_perms_from_permissions(perms_to_remove, permissions):
+    # permissions_to_remove: list of Permission objects to remove
+    # permissions: queryset of Permission objects
+    for p in perms_to_remove:
+        permissions = permissions.exclude(id=p.id)
+    return permissions
+
+
+def move_permissions_to_roles(apps, schema_editor):
+    Group = apps.get_model("galaxy", "Group")
+    GroupRole = apps.get_model("core", "GroupRole")
+    Permission = apps.get_model("auth", "Permission")
+    Role = apps.get_model("core", "Role")
+
+    group_roles = []
+
+    for i, group in enumerate(Group.objects.all()):
+        if group.name == "system:partner-engineers":
+            continue
+        permissions = group.permissions.all()
+        perms_to_remove = []
+
+        # Use Galaxy locked Roles where possible
+        for locked_perm_names, locked_rolename in GALAXY_LOCKED_ROLE_TRANSLATOR:
+            locked_perms = [Permission.objects.filter(
+                content_type__app_label=app_label, codename=codename
+            ).first() for app_label, codename in locked_perm_names]
+
+            if all(locked_perms):
```

Comment:
This looks good to me. Reasoning (maybe add as a comment). Maybe any of these permissions did not even exist by the time the migration runs. But at that time we can safely assume no group to have all of them (because they would have needed to exist).